src/Server.zig

```diff
@@ -645,94 +668,359 @@ fn handleConfiguration(server: *Server, json: std.json.Value) error{OutOfMemory}
+        server.showMessage(.Warning, "zig executable could not be found", .{});
+    }
+
+    if (server.config.enable_ast_check_diagnostics and
+        server.config.prefer_ast_check_as_child_process)
+    {
+        if (!std.process.can_spawn) {
+            log.info("'prefer_ast_check_as_child_process' is ignored because your OS can't spawn a child process", .{});
+        } else if (server.config.zig_exe_path == null) {
+            log.info("'prefer_ast_check_as_child_process' is ignored because Zig could not be found", .{});
+        }
+    }
+
+    if (server.status == .initialized) {
+        const json_message = try server.sendToClientRequest(
+            .{ .string = "semantic_tokens_refresh" },
+            "workspace/semanticTokens/refresh",
+            {},
+        );
+        server.allocator.free(json_message);
+    }
+}
+
+fn validateConfiguration(server: *Server, config: *configuration.Configuration) !void {
+    inline for (comptime std.meta.fieldNames(Config)) |field_name| {
+        const FileCheckInfo = struct {
+            kind: enum { file, directory },
+            is_accessible: bool,
+        };
+
+        @setEvalBranchQuota(2_000);
+        const file_info: FileCheckInfo = comptime if (std.mem.indexOf(u8, field_name, "path") != null) blk: {
+            if (std.mem.eql(u8, field_name, "zig_exe_path") or
+                std.mem.eql(u8, field_name, "replay_session_path") or
+                std.mem.eql(u8, field_name, "builtin_path") or
+                std.mem.eql(u8, field_name, "build_runner_path"))
+            {
+                break :blk .{ .kind = .file, .is_accessible = true };
+            } else if (std.mem.eql(u8, field_name, "record_session_path")) {
+                break :blk .{ .kind = .file, .is_accessible = false };
+            } else if (std.mem.eql(u8, field_name, "zig_lib_path")) {
+                break :blk .{ .kind = .directory, .is_accessible = true };
+            } else if (std.mem.eql(u8, field_name, "global_cache_path") or
+                std.mem.eql(u8, field_name, "build_runner_global_cache_path"))
+            {
+                break :blk .{ .kind = .directory, .is_accessible = false };
+            } else {
+                @compileError(std.fmt.comptimePrint(
+                    \\config option '{s}' contains the word 'path'.
+                    \\Please add config option validation checks above if necessary.
+                    \\If not necessary, just add a continue switch-case to ignore this error.
+                    \\
+                , .{field_name}));
+            }
+        } else continue;
+
+        const is_ok = if (@field(config, field_name)) |path| ok: {
+            if (path.len == 0) break :ok false;
+
+            if (!std.fs.path.isAbsolute(path)) {
+                server.showMessage(.Warning, "config option '{s}': expected absolute path but got '{s}'", .{ field_name, path });
+                break :ok false;
+            }
```

Comment:
Using "zig" as the zig path in order to use the zig in the `PATH` causes this check to fail, even though that works perfectly fine.

Other than that minor annoyance this PR looks good.